### PR TITLE
Starter plugin: Add E2E setup

### PR DESCRIPTION
**Pull request by brbrr**
#### Changes proposed in this Pull Request:

* Add E2E setup to Starter Plugin
* Intergate E2E tests into CI pipeline
* Update E2E setup guide

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

n/a
#### Testing instructions:

* Check the Starter Plugin E2E tests in CI. Is everything in order?
*

### CODE REVIEW

**Review by adimoldovan — COMMENTED** [review]
Looks good!
I've left some small suggestions for the readme file.

**Review by anomiex — COMMENTED** [review]
Something weird is going on in `pnpm-lock.yaml`, it's downgrading a lot of things and adding old versions of packages, and making a bunch of other weird changes too.

I'd recommend regenerating it fresh from the master version: do a master merge, then `git checkout origin/master pnpm-lock.yaml && pnpm install` to regenerate it.

File: .github/files/create-e2e-projects-matrix.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -eo pipefail
 
-PROJECTS=('{"project":"Jetpack","path":"projects/plugins/jetpack/tests/e2e","testArgs":["--grep-invert", "plugin-update"],"slackArgs":[]}' '{"project":"Jetpack update","path":"projects/plugins/jetpack/tests/e2e","testArgs":["plugin-update"],"slackArgs":[]}' '{"project":"Boost","path":"projects/plugins/boost/tests/e2e","testArgs":[],"slackArgs":[]}' '{"project":"Search","path":"projects/plugins/search/tests/e2e","testArgs":[],"slackArgs":[]}')
+PROJECTS=('{"project":"Jetpack","path":"projects/plugins/jetpack/tests/e2e","testArgs":["--grep-invert", "plugin-update"],"slackArgs":[]}' '{"project":"Jetpack update","path":"projects/plugins/jetpack/tests/e2e","testArgs":["plugin-update"],"slackArgs":[]}' '{"project":"Boost","path":"projects/plugins/boost/tests/e2e","testArgs":[],"slackArgs":[]}' '{"project":"Search","path":"projects/plugins/search/tests/e2e","testArgs":[],"slackArgs":[]}' '{"project":"Starter Plugin","path":"projects/plugins/starter-plugin/tests/e2e","testArgs":[],"slackArgs":[]}')
```
[the remark this answers is not in the document]
Comment (brbrr) [reply .github/files/create-e2e-projects-matrix.sh:5]: Yeah, it does not add any value. I added it just to set up the whole pipeline. I'll remove added line here

File: projects/plugins/starter-plugin/tests/e2e/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		"tunnel:reset": "tunnel reset",
+		"tunnel:down": "tunnel down",
+		"tunnel:write-logs": "tunnel logs output/logs/tunnel.log",
+		"pretest": "pnpm run clean",
+		"test:run": ". ./node_modules/jetpack-e2e-commons/bin/app-password.sh && playwright install && NODE_CONFIG_DIR='./config' ALLURE_RESULTS_DIR=./output/allure-results NODE_PATH=\"$PWD/node_modules\" playwright test --config=./playwright.config.cjs",
+		"slack": "NODE_CONFIG_DIR='./config' slack"
+	},
+	"devDependencies": {
+		"@playwright/test": "1.20.1",
+		"allure-playwright": "2.0.0-beta.15",
```
Comment (anomiex) [inline projects/plugins/starter-plugin/tests/e2e/package.json:35]: To match #24749 which was merged a few hours ago,
```suggestion
		"@playwright/test": "1.22.2",
		"allure-playwright": "2.0.0-beta.17",
```

**Review by adimoldovan — APPROVED** [review]

**Comment by brbrr** [comment]
r247600-wpcom

**Pull request merged**